### PR TITLE
Ci jobs

### PR DESCRIPTION
**Pull request by KyleQianliMa**
# Short description of the changes:

Created Action.yml added codecov token.

**Extra files changed:**  

In `pyproject.toml` file `ndarray` folder is no longer ignored. 

Additional tests in `src/histogram/ndarray/AbstractNdArray.py` has been moved into `tests/histogram/ndarray/NdArray_TestCase.py`. Code in `NdArray_TestCase.py` has been modified to accomodate pytest convention. Code related to tests in `src/histogram/ndarray/AbstractNdArray.py`  and `src/histogram/ndarray/NumpyNdArray.py` have been commented out to accomodate the move(i.e. referencing tests in-place has been commented out because tests are now moved into test folder).

Now if we do pytest in root directory, there should be 171 passed tests instead of 145. The extra tests come from 'test/histogram/ndarray'. This means all tests except the ones referencing 'stdVector' is covered now.

[EWM 6594](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=6594)
# Long description of the changes:

# Check list for the pull request
- [ ] I have read the [CONTRIBUTING]
- [ ] I have read the [CODE_OF_CONDUCT]
- [ ] I have added tests for my changes
- [ ] I have updated the documentation accordingly

# Check list for the reviewer
- [ ] I have read the [CONTRIBUTING]
- [ ] I have verified the proposed changes
- [ ] best software practices
    + [ ] all internal functions have an underbar, as is python standard
    + [ ] clearly named variables (better to be verbose in variable names)
    + [ ] code comments explaining the intent of code blocks
- [ ] All the tests are passing
- [ ] The documentation is up to date
- [ ] code comments added when explaining intent

# Manual test for the reviewer

# References

### CODE REVIEW

**Comment by mpatrou** [comment]
Can the src/histogram/_version.py  be excluded from the coverage report ?

**Comment by KyleQianliMa** [comment]
> Can the src/histogram/_version.py be excluded from the coverage report ?

Yes, I can exclude any files

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          channels:
+            - conda-forge
+            - mcvine
+            - mantid
+    - name: pre-commit
+      run: pre-commit run --all-files
+    - name: Install histogram
+      run: python -m pip install -e . --no-deps
+    - name: Run tests
+      run: xvfb-run --server-args="-screen 0 1920x1080x24" -a python -m pytest --cov --cov-report=xml --cov-report=term --cov-config=.coveragerc
```
Comment (KyleQianliMa) [inline .github/workflows/actions.yml:33]: --cov-config=.coveragerc calls the .coveragerc file which ignores _version.py

**Comment by KyleQianliMa** [comment]
> Can the src/histogram/_version.py be excluded from the coverage report ?

The src/histogram/_version.py has been excluded through a .coveragerc file and command line `--cov-config=.coveragerc` in action.yml. But I can't seem to include lib and bin it gives coverageWarning.

image

File: .coveragerc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[run]
+omit =
+    src/histogram/_version.py
+source =
```
Comment (mpatrou) [inline .coveragerc:4]: what about here?

Comment (KyleQianliMa) [reply .coveragerc:4]: I can add bin and lib folder here but it wouldn't run
image

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    "*.icc"
+]
+
+[tool.coverage.run]
+omit = [
+    "src/histogram/_version.py"
+    ]
+source = [
+  "src"
+]
```
Comment (KyleQianliMa) [inline pyproject.toml:63]: These lines control ignored file and source file for code coverage

File: .github/workflows/actions.yml (hunk 1 above)
Comment (mpatrou) [reply .github/workflows/actions.yml:33]: slug: neutrons/histogram-dev
check here: https://github.com/neutrons/garnet_reduction/blob/next/.github/workflows/garnetreduction.yml#L39C19-L39C50

File: .github/workflows/actions.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    - uses: actions/checkout@v4
+    - uses: conda-incubator/setup-miniconda@v3
+      with:
+        auto-update-conda: true
+        channels: conda-forge,defaults
+        mamba-version: "*"
+        environment-file: environment.yml
+        cache-environment-key: ${{ runner.os }}-env-${{ hashFiles('**/environment.yml') }}
+        cache-downloads-key: ${{ runner.os }}-downloads-${{ hashFiles('**/environment.yml') }}
+    - name: build conda package
```
Comment (mpatrou) [inline .github/workflows/actions.yml:54]: before line 54 include this
 
 - name: building python wheel
    shell: bash -l {0}
    run: |
      python -m build --wheel --no-isolation

File: .github/workflows/actions.yml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        slug: neutrons/histogram-dev
+
+  conda-build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+    - uses: actions/checkout@v4
+    - uses: conda-incubator/setup-miniconda@v3
```
Comment (mpatrou) [inline .github/workflows/actions.yml:46]: for the warning: replace line 46 conda-incubator/setup-miniconda@v3 with:
        - uses: mamba-org/setup-micromamba@v1

**Review by mpatrou — APPROVED** [review]
it looks good to me!!

**Pull request merged**